beacon-chain/simulator/service.go

```diff
@@ -156,8 +155,19 @@ func (sim *Simulator) run(delayChan <-chan time.Time, done <-chan struct{}) {
 
 			// If we have not broadcast a simulated block yet, we set parent hash
 			// to the genesis block.
+			var hash [32]byte
 			if sim.slotNum == 1 {
```

Comment:
I would consider using `BeaconChain.CanonicalHead` to get the parent hash instead. It would simplify this entire service by eliminating the need for the `lastSimulatedSessionBlock` method, the `last-simulated-block` key, and the need to save the last simulated block in `Stop`.